### ISSUE TITLE
Fails to build

### ISSUE THREAD

**Issue by thierryseegers** [robertgr991/fastDamerauLevenshtein#2]
After failing to install via `pip3`, I tried to build directly from source and I believe I just hit the same problem. This what `python3 setup.py install` produces:

```
running install
running bdist_egg
running egg_info
creating fastDamerauLevenshtein.egg-info
writing fastDamerauLevenshtein.egg-info/PKG-INFO
writing dependency_links to fastDamerauLevenshtein.egg-info/dependency_links.txt
writing top-level names to fastDamerauLevenshtein.egg-info/top_level.txt
writing manifest file 'fastDamerauLevenshtein.egg-info/SOURCES.txt'
reading manifest file 'fastDamerauLevenshtein.egg-info/SOURCES.txt'
writing manifest file 'fastDamerauLevenshtein.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.15-x86_64/egg
running install_lib
running build_ext
building 'fastDamerauLevenshtein' extension
creating build
creating build/temp.macosx-10.15-x86_64-3.8
creating build/temp.macosx-10.15-x86_64-3.8/fastDamerauLevenshtein
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c fastDamerauLevenshtein/fastDamerauLevenshtein.c -o build/temp.macosx-10.15-x86_64-3.8/fastDamerauLevenshtein/fastDamerauLevenshtein.o
fastDamerauLevenshtein/fastDamerauLevenshtein.c:4246:9: error: too many arguments to function call, expected 15, have
      16
        __pyx_empty_bytes  /*PyObject *lnotab*/
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
fastDamerauLevenshtein/fastDamerauLevenshtein.c:331:82: note: expanded from macro '__Pyx_PyCode_New'
          PyCode_New(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
          ~~~~~~~~~~                                                             ^~~~
/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/include/python3.8/code.h:122:12: note:
      'PyCode_New' declared here
PyAPI_FUNC(PyCodeObject *) PyCode_New(
           ^
1 error generated.
error: command 'clang' failed with exit status 1
```
The macro at [line 326](https://github.com/robertgr991/fastDamerauLevenshtein/blob/f6eed021ae7e99118157afb72cf0631417745037/fastDamerauLevenshtein/fastDamerauLevenshtein.c#L326) appears to imply that `PyCode_New` will take 16 parameters rather than 15 starting at v. 3.8 but [Python documentation](https://docs.python.org/3.8/c-api/code.html) appears to me to state otherwise.

**Comment by robertgr991** [comment]
Hi,

I'll take a look into it these days. Strange indeed, it's a problem with Cython, which version did you use to build it from the source?

**Comment by thierryseegers** [comment]
@robertgr991 I didn't have Cython installed at all. (I do now (v. 0.29.21) but it doesn't make a difference.) The issue I see is in this file which was generated by some earlier version Cython. It feels like it should be re-generated. (Btw, I'm speculating here. I'm not a Python expert.)

**Comment by thierryseegers** [comment]
@robertgr991 Indeed. Now that I've manually regenerated `fastDamerauLevenshtein.c` with `python3 /usr/local/lib/python3.8/site-packages/cython.py fastDamerauLevenshtein.pyx` (that's on macOS, btw), I was able to successfully invoke `python3 setup.py install` afterwards.

**Comment by robertgr991** [comment]
As I thought, it needs to be regenerated with a newer cython version. Thx for confirming this and for pointing this issue out. I will regenerate it to support newer python versions.

**Comment by thierryseegers** [comment]
Just a thought: could that generation step not happen at install time?

**Comment by robertgr991** [comment]
It's possible with wheels. I think there was an additional way, but I can't remeber it right now, I'll have to look it up.

**Comment by robertgr991** [comment]
Closed by 52d0601.

Thx
